File: frontend/src/types/match.ts
export interface SubScores {
  required_skills: number;
  preferred_skills: number;
  experience: number;
  education: number;
  projects: number;
  semantic: number;
  ats_keywords: number;
}

export interface GapAnalysis {
  strong: string[];
  partial: string[];
  missing: string[];
  transferable: string[];
}

export interface AtsAnalysis {
  ats_score: number;
  keyword_coverage_score: number;
  keywords_present: string[];
  keywords_missing: string[];
  formatting_issues: string[];
}

export type Recommendation =
  | 'STRONGLY_APPLY'
  | 'APPLY'
  | 'APPLY_WITH_CAUTION'
  | 'IMPROVE_FIRST'
  | 'DO_NOT_APPLY';

export interface MatchReport {
  id: string;
  resume_id: string;
  job_id: string;
  overall_score: number;
  recommendation: Recommendation;
  sub_scores: SubScores;
  gap_analysis: GapAnalysis;
  ats_analysis: AtsAnalysis;
}

File: backend/app/schemas/match.py
"""app/schemas/match.py — API request/response shapes for match scoring."""
from __future__ import annotations

import uuid

from pydantic import BaseModel


class MatchRequest(BaseModel):
    resume_id: uuid.UUID
    job_id: uuid.UUID


class SubScores(BaseModel):
    required_skills: float
    preferred_skills: float
    experience: float
    education: float
    projects: float
    semantic: float
    ats_keywords: float


class MatchResult(BaseModel):
    overall_score: float
    sub_scores: SubScores
    strong_matches: list[str] = []
    partial_matches: list[str] = []
    missing_skills: list[str] = []
    transferable_skills: list[str] = []
    missing_ats_keywords: list[str] = []
    recommendation: str
    explanation: str | None = None


class MatchResponse(BaseModel):
    id: uuid.UUID
    resume_id: uuid.UUID
    job_id: uuid.UUID
    result: MatchResult
